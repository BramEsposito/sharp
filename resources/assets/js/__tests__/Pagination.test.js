import { mount } from '@vue/test-utils';
import Pagination from '../components/list/Pagination.vue';

describe('pagination', ()=>{
    function createPagination(props={}) {
        return mount(Pagination, {
            context: {
                attrs: {},
                props: {
                    minPageEndButtons: 3,
                    totalRows: 100,
                    perPage: 10,
                    ...props
                }
            }
        });
    }

    test('mount pagination', ()=>{
        expect(createPagination().html()).toMatchSnapshot();
    });

    test('mount pagination without end buttons', ()=>{
        expect(createPagination({ totalRows: 20 }).html()).toMatchSnapshot();
    });
});