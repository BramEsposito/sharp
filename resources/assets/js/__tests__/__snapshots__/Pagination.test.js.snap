// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination mount pagination 1`] = `
<ul role="menubar" aria-disabled="false" aria-label="Pagination" class="SharpPagination pagination b-pagination pagination-md">
	<li role="none presentation" aria-hidden="true" class="page-item disabled">
		<span class="page-link">
			<i class="fa fa-angle-double-left" aria-hidden="true"></i>
		</span>
	</li>
	<li role="none presentation" aria-hidden="true" class="page-item disabled">
		<span class="page-link">
			<i class="fa fa-angle-left" aria-hidden="true"></i>
		</span>
	</li>
	<!---->
	<li role="none presentation" class="page-item active">
		<a role="menuitemradio" tabindex="0" aria-label="Goto page 1" aria-checked="true" aria-posinset="1" aria-setsize="10" href="#" target="_self" class="page-link btn-primary">1</a>
	</li>
	<li role="none presentation" class="page-item">
		<a role="menuitemradio" tabindex="-1" aria-label="Goto page 2" aria-checked="false" aria-posinset="2" aria-setsize="10" href="#" target="_self" class="page-link">2</a>
	</li>
	<li role="none presentation" class="page-item">
		<a role="menuitemradio" tabindex="-1" aria-label="Goto page 3" aria-checked="false" aria-posinset="3" aria-setsize="10" href="#" target="_self" class="page-link">3</a>
	</li>
	<li role="none presentation" class="page-item d-none d-sm-flex">
		<a role="menuitemradio" tabindex="-1" aria-label="Goto page 4" aria-checked="false" aria-posinset="4" aria-setsize="10" href="#" target="_self" class="page-link">4</a>
	</li>
	<li role="separator" class="page-item disabled d-none d-sm-flex">
		<span class="page-link">â€¦</span>
	</li>
	<li role="none presentation" class="page-item">
		<a role="menuitem" tabindex="-1" aria-label="Goto next page" href="#" target="_self" class="page-link">
			<span aria-hidden="true">
				<i class="fa fa-angle-right" aria-hidden="true"></i>
			</span>
		</a>
	</li>
	<li role="none presentation" class="page-item">
		<a role="menuitem" tabindex="-1" aria-label="Goto last page" href="#" target="_self" class="page-link">
			<span aria-hidden="true">
				<i class="fa fa-angle-double-right" aria-hidden="true"></i>
			</span>
		</a>
	</li>
</ul>
`;

exports[`pagination mount pagination without end buttons 1`] = `
<ul role="menubar" aria-disabled="false" aria-label="Pagination" class="SharpPagination pagination b-pagination pagination-md">
	<!---->
	<li role="none presentation" aria-hidden="true" class="page-item disabled">
		<span class="page-link">
			<i class="fa fa-angle-left" aria-hidden="true"></i>
		</span>
	</li>
	<!---->
	<li role="none presentation" class="page-item active">
		<a role="menuitemradio" tabindex="0" aria-label="Goto page 1" aria-checked="true" aria-posinset="1" aria-setsize="2" href="#" target="_self" class="page-link btn-primary">1</a>
	</li>
	<li role="none presentation" class="page-item">
		<a role="menuitemradio" tabindex="-1" aria-label="Goto page 2" aria-checked="false" aria-posinset="2" aria-setsize="2" href="#" target="_self" class="page-link">2</a>
	</li>
	<!---->
	<li role="none presentation" class="page-item">
		<a role="menuitem" tabindex="-1" aria-label="Goto next page" href="#" target="_self" class="page-link">
			<span aria-hidden="true">
				<i class="fa fa-angle-right" aria-hidden="true"></i>
			</span>
		</a>
	</li>
	<!---->
</ul>
`;
